### PR TITLE
docs: added responsePostProcessor, reorganized backdoor for narrative flow

### PR DESCRIPTION
**Pull request by amzn-chris-subagio**
Added responsePostProcessor documentation to the section on modifying responses.
Organized backdoor.md by cognitive load, with the heavier context object info last. 
Moved the DBKey override information next to the conversation about database storage in the variables pages instead.

### CODE REVIEW

File: docs/book/backdoor.md [hunk 1]
```diff
@@ -27,17 +219,26 @@ launch
 printObject = function(obj) { console.log(JSON.stringify(obj)); }
 ```
 
-This Litexa `context` is recreated for every new skill request, and contains the following data:
+The Litexa `context` is recreated for every new skill request, and contains the following data:
 
-1. incoming request data (e.g. session attributes)
-2. pending response data (e.g. pending output speech)
+1. incoming request data (e.g. the Alexa skill session attributes)
+2. accumulated response data (e.g. an array of all the say statements hit so far)
```
Comment (ayrc) [inline docs/book/backdoor.md]: nitpick: say -> `say`

File: docs/book/backdoor.md [hunk 2]
```diff
@@ -113,33 +314,19 @@ The full incoming request, along with session and system information, is availab
 :::tip Request Data Shortcuts
 There are some useful request data shortcuts available in Litexa:
 
-* `context.event.request` is available as [$request](/reference/#request).
-* `context.event.request.intent.slots` values are populated directly in
-[slot variables](/reference/#slot-variables). For instance:
+* `context.event.request` is available in Litexa code as the request variable [$request](/reference/#request).
+* `context.requestId` is a shortcut to `context.event.request.requestId`
```
Comment (ayrc) [inline docs/book/backdoor.md]: nitpick: period at the end

File: docs/book/expressions.md [hunk 3]
```diff
@@ -311,6 +312,40 @@ Otherwise, if you want to deploy multiple skills where most of their
...
 `DEPLOY` object configurations, thereby avoiding code duplication.
 
+
+## Customizing the Litexa DB Key
+
+For variables that have database storage, Litexa maintains a single document in the
+backing database, by default keyed to the Alexa request's deviceId field, meaning that
+the data will be preserved for the skill running on the same Alexa device only.
+
+Should prefer to key on the userId instead (same skill data, no matter which device
```
Comment (ayrc) [inline docs/book/expressions.md]: Should prefer -> Should you prefer

File: docs/book/expressions.md [hunk 4]
```diff
@@ -311,6 +312,40 @@ Otherwise, if you want to deploy multiple skills where most of their
 logic is the same, you can keep them as one Litexa project with different
 `DEPLOY` object configurations, thereby avoiding code duplication.
 
+
+## Customizing the Litexa DB Key
```
Comment (ayrc) [inline docs/book/expressions.md]: I don't think this is the right place to put this - this isn't related at all to variable types. Can you move it back?

File: docs/book/backdoor.md [hunk 5]
```diff
@@ -253,178 +440,12 @@ launch
...
-
-    ```json
-    {
-      "name": "myFirstProject",
-      "deployments": { ... },
-      "directiveWhitelist": [
-        "AudioPlayer.Play"
-      ]
-    }
-    ```
```
Comment (ayrc) [inline docs/book/backdoor.md]: Is this not supported anymore? or are we discouraging this in favor of extensions? I think we should keep this in here or move it to the `directive` language reference.

Comment (amzn-chris-subagio) [reply docs/book/backdoor.md]: It does, it moved up in the document before the context section.

File: docs/book/expressions.md (hunk 4 above)
Comment (amzn-chris-subagio) [reply docs/book/expressions.md]: Updating to make it clearer how this belongs next to the description of storage scopes.

File: docs/book/backdoor.md [hunk 6]
```diff
@@ -7,14 +7,206 @@ code.
...
+*Not Yet Available*
+
+There may be times when manual intervention in the model building process is necessary,
+e.g. you may want to inject some other procedural mechanism for creating values that
+doesn't fit Litexa's slotBuilder scheme. If you have a case like this, please reach
+out to us with a proposal for where you feel it would be natural to inject it!
+
+## Accepting Novel Events
+
+To accept named events that Litexa does not yet define in it's core, you can write
```
Comment (ayrc) [inline docs/book/backdoor.md]: it's -> its

File: docs/book/backdoor.md [hunk 7]
```diff
@@ -7,14 +7,206 @@ code.
...
+        "image": {
+          "smallImageUrl": "https://there.com/rose-small.png",
+          "largeImageUrl": "https://there.com/rose-large.png"
+        }
+      }
+    }
+  }
+```
+
+
```
Comment (ayrc) [inline docs/book/backdoor.md]: remove extra spacing

File: docs/book/backdoor.md [hunk 8]
```diff
@@ -7,14 +7,206 @@ code.
...
+e.g. you may want to inject some other procedural mechanism for creating values that
+doesn't fit Litexa's slotBuilder scheme. If you have a case like this, please reach
+out to us with a proposal for where you feel it would be natural to inject it!
+
+## Accepting Novel Events
+
+To accept named events that Litexa does not yet define in it's core, you can write
+an extension module, where you indicate the new events in the
+`compiler.valideEventNames` list. [Litexa extension](extensions.html#_2-compiler-extension)
+
```
Comment (ayrc) [inline docs/book/backdoor.md]: remove extra spacing

**Pull request merged**